plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -688,7 +697,14 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
                      self->log_trx_results( trx, nullptr, ex, 0, start, is_transient );
                      next( std::move(ex) );
                      self->_idle_trx_time = fc::time_point::now();
-                     self->_time_tracker.add_fail_time(self->_idle_trx_time - start);
+                     auto dur = self->_idle_trx_time - start;
+                     if ( is_transient ) {
+                        // this function can only be on main thread.
+                        // transient time includes both success and fail time
+                        self->_time_tracker.add_transient_time(dur);
+                     } else {
+                        self->_time_tracker.add_fail_time(dur);
+                     }
```

Comment:
I think this code would be simpler if instead you had `add_fail_time` and `add_success_time` take a `bool is_transient` so that this all becomes:
`self->_time_tracker.add_fail_time(self->_idle_trx_time - start, is_transient);`

Reply:
oh no, we cannot do this, as for transient transactions, we don't distinguish between fail and success.

Reply:
Sure you can, something like:
```
   void add_fail_time( const fc::microseconds& fail_time, bool is_transient ) {
      if (is_transient) {
         transient_trx_time += time;
         ++transient_trx_num;
      } else {
         trx_fail_time += fail_time;
         ++trx_fail_num;
      }
   }

```
And similar change in `add_success_time`.

Reply:
Do you think this makes add_fail_time and add_success_time more complicated?

Reply:
Yes, but it moves the complication into one place instead of duplicating the `if`

Reply:
But on the other hand, this hides how transient is handled. Will change.

Reply:
It should be the time_trackers job to know what it is tracking. Then if we decide to also track fail/success separate it will be a trivial change.